### DISCUSSION TITLE
Passing value inside register object properties

### DISCUSSION THREAD

**Discussion by nenadfilipovic** [react-hook-form/react-hook-form#5540] [Q&A] [answered]
Hi, why library does not accept my value prop I pass into register object like: {...register("somename", {value: somevalue"}.

If i pass value inside input element everything is ok but that way I do not have autocomplete for value field.

When I log submitted value, and my checkbox or radio have custom value inside register, log is always "on" for that field.

Here is api example:
https://react-hook-form.com/api/useform/register
register('firstName', { value: 'bill' })

**Comment by bluebill1049** [comment, marked as answer]
provide a codesanbox

**Reply by nenadfilipovic** [reply]
Here you go https://codesandbox.io/s/naughty-water-ny8rb?file=/src/App.tsx, with only 1 radio input it works ok but when I put another it gives me problem like this `boundsOptions: "on"`. Maybe problem exist because I give them same name?

**Reply by bluebill1049** [reply]
value should not pass inside the register method, leave it at the input level.

**Reply by nenadfilipovic** [reply]
I have no type safety at input level, or am I?
